script/Deploy.s.sol

```diff
@@ -8,14 +10,15 @@ contract DeployScript is Script {
         bytes32 bNode = vm.ensNamehash(baseEns);
 
         vm.startBroadcast();
-        COWShed cowshed = new COWShed();
-        COWShedFactory factory = new COWShedFactory(address(cowshed), bName, bNode);
+        COWShed cowshed = new COWShed{salt: "v1.0.0"}();
+        COWShedFactory factory = new COWShedFactory{salt: "v1.0.0"}(address(cowshed), bName, bNode);
```

Comment:
tip: here instead of declaring the version (a very tempting thing to do), we could actually just declare an empty string. Then, if the code changes (i.e. a new version), it will still deploy to a new address as the contract code is different, but it will remain deterministic.
```suggestion
        COWShed cowshed = new COWShed{salt: ""}();
        COWShedFactory factory = new COWShedFactory{salt: ""}(address(cowshed), bName, bNode);
```